evalml/tests/component_tests/test_time_series_imputer.py

```diff
@@ -0,0 +1,506 @@
+    )
+
+    imputer = TimeSeriesImputer(categorical_impute_strategy="backwards_fill")
+    transformed, _ = imputer.fit_transform(X, y)
+    assert_frame_equal(transformed, expected, check_dtype=False)
+
+
+def test_categorical_and_numeric_input(imputer_test_data):
+    X = imputer_test_data
+    y = pd.Series([0, 0, 1, 0, 1])
+    imputer = TimeSeriesImputer()
+    imputer.fit(X, y)
+    transformed, _ = imputer.transform(X, y)
+    expected = pd.DataFrame(
+        {
+            "dates": pd.date_range("01-01-2022", periods=20),
+            "categorical col": pd.Series(
+                ["zero", "one", "two", "zero", "two"] * 4, dtype="category"
+            ),
+            "int col": [0, 1, 2, 0, 3] * 4,
+            "object col": pd.Series(["b", "b", "a", "c", "d"] * 4, dtype="category"),
+            "float col": [0.0, 1.0, 0.0, -2.0, 5.0] * 4,
+            "bool col": [True, False, False, True, True] * 4,
+            "categorical with nan": pd.Series(
+                ["1", "1", "0", "0", "3"] + ["3", "1", "0", "0", "3"] * 3,
+                dtype="category",
+            ),
+            "int with nan": [1, 1, 0, 0, 1] * 4,
+            "float with nan": [0.0, 1.0, 0, -1.0, 0.0] * 4,
+            "object with nan": pd.Series(
+                ["b", "b", "b", "c", "c"] * 4, dtype="category"
+            ),
+            "bool col with nan": pd.Series(
+                [True, True, False, False, True] * 4, dtype="category"
+            ),
+            "natural language col": pd.Series(
+                ["cats are really great", "don't", "believe", "me?", "well..."] * 4,
+                dtype="string",
+            ),
+        }
+    )
+    assert_frame_equal(transformed, expected, check_dtype=False)
+
+    imputer = TimeSeriesImputer(
+        numeric_impute_strategy="forwards_fill",
+        categorical_impute_strategy="forwards_fill",
+    )
+    transformed, _ = imputer.fit_transform(X, y)
+    expected["float with nan"] = [0.0, 1.0, 1.0, -1.0, 0.0] * 4
+    assert_frame_equal(transformed, expected, check_dtype=False)
+
+
+def test_impute_target():
+    X = pd.DataFrame({"dates": pd.date_range("01-01-2022", periods=20)})
+    y = pd.Series([0, 1, 0, 1, np.nan] * 4)
+
+    imputer = TimeSeriesImputer(target_impute_strategy="forwards_fill")
+    _, y_t = imputer.fit_transform(X, y)
+    expected = pd.Series([0, 1, 0, 1, 1] * 4)
+    assert_series_equal(y_t, expected, check_dtype=False)
+
+    imputer = TimeSeriesImputer(target_impute_strategy="backwards_fill")
+    _, y_t = imputer.fit_transform(X, y)
+    expected = pd.Series([0, 1, 0, 1, 0] * 3 + [0, 1, 0, 1, 1])
+    assert_series_equal(y_t, expected, check_dtype=False)
+
+    imputer = TimeSeriesImputer(target_impute_strategy="interpolate")
+    _, y_t = imputer.fit_transform(X, y)
+    expected = pd.Series([0, 1, 0, 1, 0.5] * 3 + [0, 1, 0, 1, 1])
+    assert_series_equal(y_t, expected, check_dtype=False)
+
+
+def test_drop_all_columns(imputer_test_data):
```

Comment:
OK, I think this probably addresses my previous comment about explicitly asserting that these columns were not in the results.  I think this is fine then, maybe just hyper nit on the name `test_imputer_drops_columns_with_all_nan`.  I know it's a bit wordier but I think it's a little clearer.